e2e/tests/components/dashboard-widgets/dashboard-widgets.e2e-spec.ts

```diff
@@ -0,0 +1,72 @@
+import { browser, ElementFinder } from 'protractor';
+import { DashboardWidgetsPage } from './dashboard-widgets.po.spec';
+
+describe('Dashboard Widgets Tests', () => {
+
+    let page: DashboardWidgetsPage;
+    let widgetActions: ElementFinder;
+    let widgetEnum: ElementFinder;
+    let widgetTable: ElementFinder;
+    let widgetText: ElementFinder;
+
+    beforeEach(async () => {
+        page = new DashboardWidgetsPage();
+        await page.getPage();
+
+        // set the browser window to a specific size to ensure consistency
+        await browser.driver.manage().window().setSize(1320, 800);
+
+        widgetActions = await page.getWidget('actions-widget');
+        widgetEnum = await page.getWidget('enum-widget');
+        widgetTable = await page.getWidget('table-widget');
+        widgetText = await page.getWidget('text-widget');
+    });
+
+    // restore the window to its original size after all these tests have run
+    afterAll(async () => {
+        await browser.driver.manage().window().setSize(800, 600);
+    });
+
+    it('should have correct initial states', async () => {
```

Comment:
Can you add proper tests for the individual widget functionality instead of these which are just duplicates of existing dashboard e2e tests?

Reply:
Although these tests are copied, I would not like to remove them, since they check the initial layout and the move functionality. We will add more tests, of course.